### ISSUE TITLE
Incorrect mesh normals in shaders

### ISSUE THREAD

**Issue by Master-J** [godotengine/godot#68326]
### Godot version

4.0 beta 4

### System information

Windows 10, Vulkan

### Issue description

I'm currently trying to port some outline shaders i made in Godot 3.x to Godot 4 and noticed that the normals of both primitive and imported meshes seems to be incorrect when used in a shader's vertex function. 

The issue makes the outline appear "pinned" on the object's axes (X/Y/Z).

At first, i thought that there may be an issue with my shader, so i tried making an outline using the "Grow" feature of the StandardMaterial3D and ended up with the same result.

Here's how it looks like in Godot 4 beta 4 :
![image](https://user-images.githubusercontent.com/54776415/200170544-38d21c3a-71f6-4bce-8ef4-9646fd4ea0ec.png)

And how it should look (using Godot 3.5 and the same scene setup/shader) : 
![image](https://user-images.githubusercontent.com/54776415/200170630-61ab78b6-739f-442f-8f05-3e9986c39549.png)

### Steps to reproduce

Create a new 3d scene using Godot 4 beta 4

Add a new MeshInstance3D with a primitive mesh (the issue is easy to spot using spheres, capsules and toruses)

Give the MeshInstance3D either : 
- a new StandardMaterial3D with Grow enabled and some thickness
- a new ShaderMaterial with this shader code 
```
shader_type spatial;
render_mode cull_front, unshaded;

void vertex()
{
	VERTEX += NORMAL * 0.3;
}

void fragment()
{
	ALBEDO = vec3(1.0, 0.0, 0.0);
}
```

**Comment by fire** [comment]
@clayjohn Is this related to octahedral compression?

**Comment by Calinou** [comment]
I remember seeing an issue that reported the exact same problem, but I can't find it right now.

**Comment by clayjohn** [comment]
> @clayjohn Is this related to octahedral compression?

It doesn't look like it. Octahedral compression problems usually come from loading meshes that were saved without octohedral compression as if they had octahedral compression enabled. In this case, the user is using PrimitiveMeshes which are always generated at load time. Which means they always have octahedral compression enabled.

**Comment by Master-J** [comment]
I did some more research on this problem.
It looks like the NORMAL built-in vector isn't normalized.

If i use the normalize() function on NORMAL the result looks how it should.

**Comment by clayjohn** [comment]
Ah, yep that looks like the problem, in 3.x we always normalized the normal value after decompression:
https://github.com/godotengine/godot/blob/92aedd506375356d3323d8896a02d6c9bc11661e/drivers/gles3/shaders/scene.glsl#L293-L298

But it looks like in master we don't:
https://github.com/godotengine/godot/blob/f814e15c7f60a685e8b3623fc93231c18ccd3627/servers/rendering/renderer_rd/shaders/forward_clustered/scene_forward_clustered.glsl#L61-L66

**Comment by Master-J** [comment]
I did some more research on this problem.
It looks like the NORMAL built-in vector isn't normalized.

If i use the normalize() function on NORMAL the result looks how it should.

**Comment by masaaldosey** [comment]
Hi, would like to take up this issue 😄

**Comment by Ayush-singla27** [comment]
@masaaldosey are you still working on this issue? i would also like to give it a go.

**Comment by akien-mga** [comment]
Fixed by #68830.
